Wire protobuf generation in codegen plugin

Baccata opened the pull request:
In preparation for built-in protobuf support, we're wiring in smithytranslate in order to
generate protobuf specifications at build time from smithy specifications.

In addition, we're modifying the build to ensure that bootstrapped resources get cleaned up by SBT

Depends on https://github.com/disneystreaming/smithy4s/pull/1437

## PR Checklist (not all items are relevant to all PRs)

- [x] Added bootstrapped code + smoke tests (when the rendering logic is modified)
- [x] Added build-plugins integration tests (when reflection loading is required at codegen-time)
- [x] Added documentation
- [x] Updated changelog

Baccata committed:
Wire protobuf generation in codegen plugin

In preparation for built-in protobuf support,
we're wiring in smithytranslate in order to
generate protobuf specifications at build time
from smithy specifications.

In addition, we're modifying the build to ensure
that bootstrapped resources get cleaned up by SBT

Baccata committed:
Added changelog entry and documentation for protobuf conversion

Baccata committed:
Add smithytranslate namespace to reserved

kubukoz commented on sampleSpecs/protoTest.smithy:
```diff
@@ -197,15 +177,85 @@ structure InlinedUnionWrapper {
+    openInt: OpenInt
+}
+
+// NB : order of enum values generated by smithy4s is not
```
**question:** is not what?

Baccata replied:
This what meant to say that the ordering of enum values is not guaranteed. I've removed the comment in favour of this PR : https://github.com/disneystreaming/smithy4s/pull/1437

Baccata committed:
Respect enum values ordering

Amend rendering logic to help guarantee the ordering of enum values
in rendered enumeration.

Add a property test to check it.

Baccata committed:
Ensure that dynamic enum schemas respect original ordering

Baccata committed:
Amend wording of changelog

Baccata committed:
Merge branch 'guarantee-enum-ordering' into add-protobuf-generation

Baccata committed:
Remove obsolete comment and explicit proto indexes from enums

Baccata committed:
Make Scala 3 happy

Baccata committed:
Make scala 3 happy

Baccata committed:
Merge branch 'guarantee-enum-ordering' into add-protobuf-generation

Baccata committed:
Added scripted test for protobuf generation

daddykotex commented:
well done

Baccata committed:
Bump smithy-translate to 0.5.1
